libs/remix-ui/run-tab/src/lib/actions/index.ts

```diff
@@ -0,0 +1,540 @@
+
+  compiler.visitContracts((contract) => {
+    contracts.push(contract)
+  })
+  return contracts
+}
+
+export const getSelectedContract = (contractName: string, compilerAtributeName: string) => {
+  if (!contractName) return null
+  const compiler = plugin.compilersArtefacts[compilerAtributeName]
+
+  if (!compiler) return null
+
+  const contract = compiler.getContract(contractName)
+
+  return {
+    name: contractName,
+    contract: contract,
+    compiler: compiler,
+    abi: contract.object.abi,
+    bytecodeObject: contract.object.evm.bytecode.object,
+    bytecodeLinkReferences: contract.object.evm.bytecode.linkReferences,
+    object: contract.object,
+    deployedBytecode: contract.object.evm.deployedBytecode,
+    getConstructorInterface: () => {
+      return txHelper.getConstructorInterface(contract.object.abi)
+    },
+    getConstructorInputs: () => {
+      const constructorInteface = txHelper.getConstructorInterface(contract.object.abi)
+
+      return txHelper.inputParametersDeclarationToString(constructorInteface.inputs)
+    },
+    isOverSizeLimit: () => {
+      const deployedBytecode = contract.object.evm.deployedBytecode
+
+      return (deployedBytecode && deployedBytecode.object.length / 2 > 24576)
+    },
+    metadata: contract.object.metadata
+  }
+}
+
+const getCompilerContracts = () => {
+  return plugin.compilersArtefacts.__last.getData().contracts
+}
+
+const terminalLogger = (view: JSX.Element) => {
+  plugin.call('terminal', 'logHtml', view)
+}
+
+const getConfirmationCb = (confirmDialogContent: (
+  tx: Tx, network:
+  Network, amount: string,
+  gasEstimation: string,
+  gasFees: (maxFee: string, cb: (txFeeText: string, priceStatus: boolean) => void) => void,
+  determineGasPrice: (cb: (txFeeText: string, gasPriceValue: string, gasPriceStatus: boolean) => void) => void
+  ) => JSX.Element) => {
+  // this code is the same as in recorder.js. TODO need to be refactored out
+  const confirmationCb = (network, tx, gasEstimation, continueTxExecution, cancelCb) => {
+    if (network.name !== 'Main') {
+      return continueTxExecution(null)
+    }
+    const amount = plugin.blockchain.fromWei(tx.value, true, 'ether')
+    const content = confirmDialogContent(tx, network, amount, gasEstimation, plugin.blockchain.determineGasFees(tx), plugin.blockchain.determineGasPrice.bind(plugin.blockchain))
+
+    dispatch(displayNotification('Confirm transaction', content, 'Confirm', 'Cancel', () => {
+      plugin.blockchain.config.setUnpersistedProperty('doNotShowTransactionConfirmationAgain', plugin.REACT_API.confirmSettings)
+      // TODO: check if this is check is still valid given the refactor
+      if (!plugin.REACT_API.gasPriceStatus) {
+        cancelCb('Given transaction fee is not correct')
+      } else {
+        continueTxExecution({ maxFee: plugin.REACT_API.maxFee, maxPriorityFee: plugin.REACT_API.maxPriorityFee, baseFeePerGas: plugin.REACT_API.baseFeePerGas, gasPrice: plugin.REACT_API.gasPrice })
```

Comment:
@yann300 Suggested including address as a field passed in to `continueTxExecution` to fix mainnet deployment error.